stdlib/src/builtin/simd.mojo

```diff
@@ -2275,39 +2276,87 @@ struct SIMD[type: DType, size: Int = simdwidthof[type]()](
         return self.reduce[mul_reduce_body, size_out]()
 
     @always_inline
-    fn reduce_and(self) -> Bool:
-        """Reduces the boolean vector using the `and` operator.
+    fn reduce_and[size_out: Int = 1](self) -> SIMD[type, size_out]:
+        """Reduces the vector using the bitwise and `&` operator.
+
+        Parameters:
+            size_out: The width of the reduction.
 
         Constraints:
-            The element type of the vector must be boolean.
+            `size_out` must not exceed width of the vector.
+            The element type of the vector must be integer or boolean.
 
         Returns:
-            True if all element in the vector is True and False otherwise.
+            The bitwise and `&` of all vector elements.
```

Comment:
```suggestion
            The bitwise AND of all vector elements.
```

Reply:
I'm gonna make it even simpler, since that statement isn't always accurate anyways.

Reply:
if it's ok: "returns the reduced vector" makes sense in all cases